chsdi/templates/htmlpopup/gisgeol.mako

```diff
@@ -1,11 +1,31 @@
 <%inherit file="base.mako"/>
-<%!
+
+<%def name="table_body(c,lang)">
+<%
     import markupsafe
+    import glob,os
+    api_url = h.make_api_url(request, True)
+
     def br(text):
         return text.replace('\n', markupsafe.Markup('<br />'))
+
+    def hbytes(num):
+        for x in ['bytes','KB','MB','GB']:
+            if num < 1024.0:
+                return "%3.1f %s" % (num, x)
+            num /= 1024.0
+        return "%3.1f %s" % (num, 'TB')
+    
+    def findfiles(text):
+        result = []
+        for file in glob.glob('/var/local/cartoweb/downloads/gisgeol/%s*' % text ):
```

Comment:
+1 for that.

Might be worth writing a helper class wrapping the zadara specific code, instead of having it in a mako. It could be used in other services as well (maybe when we need to provide upload service, list service, etc). Helper would allow:

- listing of files
- getting file urls
- etc...